packages/server-core/src/media/file-browser/file-browser.class.ts

```diff
@@ -325,6 +325,9 @@ export class FileBrowserService
     const oldFullPath = `${data.oldPath}${data.oldName}/`.replace(TRAILING_SLASH_REGEX, '')
     const newFullPath = data.newPath.replace(TRAILING_SLASH_REGEX, '')
 
+    const key = path.join(newFullPath, data.newName)
+    if (data.unique) data.newName = await (await ensureUniqueName(this.app, key)).split('/').pop()!
```

Comment:
what is the condition upon not wanting a unique file name? Overriding an existing file?